src/main/Duke/virtue/VirtueTaskList.java

```diff
@@ -0,0 +1,168 @@
+package virtue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class VirtueTaskList {
+    // The task list.
+    private List<VirtueTask> taskList;
+
+    public VirtueTaskList() {
+        taskList = new ArrayList<>();
+    }
+
+    // Gets the number of tasks in the task list.
+    private int numTasks() {
+        return taskList.size();
+    }
+
+    // Gets the i-th task from the task list, where i is the index.
+    private VirtueTask getTask(int index) {
+        return taskList.get(index - 1);
+    }
+
+    // Adds a todo task to the task list.
+    private void addTodo(String description) {
+        Todo todo = new Todo(description);
+        taskList.add(todo);
+        int numTasks = numTasks();
+        String sOrNone = numTasks == 1 ? "" : "s";
+        Virtue.printHorizontalLine();
+        Virtue.printWithIndent(" Got it. I've added this task:");
+        Virtue.printWithIndent("   " + todo);
+        Virtue.printWithIndent(" Now you have " + numTasks + " task" + sOrNone + " in the list.");
+        Virtue.printHorizontalLine();
+    }
+
+    // Adds a deadline task to the task list.
+    private void addDeadline(String description, String by) {
```

Comment:
the code for updating the user about task addition and completion is repeated across 'addTodo', 'addDeadline', 'addEvent'. It might be useful to abstract this code into a method that accepts parameters to customize the message